include/boost/math/interpolators/detail/2d_akima_detail.hpp

```diff
@@ -0,0 +1,159 @@
+// Copyright Matt Borland, 2021
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Source: https://www.its.bldrdoc.gov/DownloadPublications.ashx?id=OT-75-70.pdf
+
+#ifndef BOOST_MATH_INTERPOLATORS_2D_AKIMA_DETIAL_HPP
+#define BOOST_MATH_INTERPOLATORS_2D_AKIMA_DETIAL_HPP
+
+#include <type_traits>
+#include <stdexcept>
+#include <utility>
+#include <algorithm>
+#include <iostream>
+#include <cstddef>
+#include <cmath>
+
+namespace boost { namespace math { namespace interpolators { namespace detail {
+
+template <typename Matrix, typename VectorType = typename Matrix::value_type, typename Real = typename VectorType::value_type>
+class akima_2d_uniform_impl
+{
+private:    
+    const Matrix data_;
+    const std::size_t n_interpolation_points_;
```

Comment:
Wait, do I have to decide beforehand how many points I want to interpolate at?

Reply:
That was how the author's implementation works, but honestly I have not made it far enough to know if it's a hard requirement or not. I am implementing the steps of the procedure as I go down the paper as non-member functions so there is flexibility in interface. Same generally goes for your other comments.

Reply:
Yeah, that's a restriction of Fortran, not of this particular method. So your goal is to first do a faithful tranlation of the paper and the C++ it? (If so I'll shut up for a while.)

Reply:
He details a step by step procedure using equations that I am using to implement this.  There is a not so great scan copy of his code that is written in FORTRAN 66 which I would prefer not to translate. I am sure there are efficiencies that can be gained from the last 50 years.